vsintegration/src/FSharp.Editor/Common/Pervasive.fs

```diff
@@ -155,9 +157,13 @@ type AsyncMaybeBuilder () =
             x.Zero ()
 
     [<DebuggerStepThrough>]
-    member x.For (sequence : seq<_>, body : 'T -> Async<unit option>) : Async<_ option> =
-        x.Using (sequence.GetEnumerator (), fun enum ->
-            x.While (enum.MoveNext, x.Delay (fun () -> body enum.Current)))
+    member x.For (sequence : seq<_>, body : 'T -> Async<unit option>) : Async<unit option> =
+        async {
+            for item in sequence do
+               let! _result = body item
```

Comment:
Do we care that each thing happens in order before the next is kicked off, or can we do something like this:

`body item |> ignore`?

Reply:
I couldn't do `body item |> ignore`

Reply:
You also can't do `do! body item`

Reply:
`option` makes this a bit weird

Reply:
What was wrong with the original code?  Now you've fixed `Using` it looks like the original code would be ok?

Reply:
My intention was to go through the same path as Async rather than what it was. They code may have been correct before, but I decided to do it this way.

Reply:
The old code is the normal definition of "For" for a computation exoression, so I'd recommend just keeping it unchanged. It depends on Using being correctly implemented, which it wasn't, but now you've fixed it the old code is best.

Reply:
Makes sense. I'll put it back.